From 36012a8e42c8486dec379b6f4ae144d19f3f5101 Mon Sep 17 00:00:00 2001
From: Tobias Wiesner <wiesner@lnm.mw.tum.de>
Date: Tue, 25 Nov 2014 23:33:25 +0100
Subject: [PATCH 08/16] deactivate two routines in LINALG::UTILS which
 directly depend on BLAS/LAPACK. One has to make sure
 that these routines are available or replace the
 functions with equivalent implementations.

---
 src/linalg/linalg_utils.cpp |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/src/linalg/linalg_utils.cpp b/src/linalg/linalg_utils.cpp
index 5552c04..c63cbcd 100644
--- a/src/linalg/linalg_utils.cpp
+++ b/src/linalg/linalg_utils.cpp
@@ -551,6 +551,10 @@ double LINALG::DeterminantLU(const Epetra_SerialDenseMatrix& A)
  *----------------------------------------------------------------------*/
 void LINALG::SymmetricInverse(Epetra_SerialDenseMatrix& A, const int dim)
 {
+#if 1
+  dserror("Check availability of BLAS/LAPACK routines for SymmetricInverse routine for a dense matrix!");
+  return;
+#else
   if (A.M() != A.N())
     dserror("Matrix is not square");
   if (A.M() != dim)
@@ -578,6 +582,7 @@ void LINALG::SymmetricInverse(Epetra_SerialDenseMatrix& A, const int dim)
     for (int j = 0; j < i; ++j)
       A(j, i) = A(i, j);
   return;
+#endif
 }
 
 /*----------------------------------------------------------------------*
@@ -782,6 +787,10 @@ void LINALG::SymmetricEigen(Epetra_SerialDenseMatrix& A,
 double LINALG::GeneralizedEigen(Epetra_SerialDenseMatrix& A,
     Epetra_SerialDenseMatrix& B)
 {
+#if 1
+  dserror("Check availability of BLAS/LAPACK routines for GeneralizedEigen routine for a dense matrix!");
+  return 0.0;
+#else
   Epetra_SerialDenseMatrix tmpA(A);
   Epetra_SerialDenseMatrix tmpB(B);
 
@@ -991,6 +1000,7 @@ double LINALG::GeneralizedEigen(Epetra_SerialDenseMatrix& A,
     }
   }
   return maxlambda;
+#endif
 }
 
 /*----------------------------------------------------------------------*
-- 
1.7.7.6

